src/Razor/src/Microsoft.AspNetCore.Razor.ProjectEngineHost/Utilities/RazorProjectInfoFactory.cs

```diff
@@ -114,6 +113,7 @@ private static RazorConfiguration ComputeRazorConfigurationOptions(AnalyzerConfi
         configurationName ??= "MVC-3.0"; // TODO: Source generator uses "default" here??
 
         globalOptions.TryGetValue("build_property.RootNamespace", out var rootNamespace);
+        defaultNamespace = rootNamespace ?? "ASP"; // TODO: Source generator does this. Do we want it?
```

Comment:
Now that `RazorConfiguration.RootNamespace` has this value, is the `out string defaultNamespace` parameter still needed? It looks like this could be simplified a bit further.

Reply:
You were right, I updated all of this. Sadly still need an output parameter, and I need to log an issue to follow up with respect to that.